src/encoding.ts

```diff
@@ -90,50 +91,50 @@ export interface Encoding<F> {
    * 1) For fine-grained control over both fill and stroke colors of the marks, please use the `fill` and `stroke` channels.  If either `fill` or `stroke` channel is specified, `color` channel will be ignored.
    * 2) See the scale documentation for more information about customizing [color scheme](https://vega.github.io/vega-lite/docs/scale.html#scheme).
    */
-  color?: FieldDefWithCondition<MarkPropFieldDef<F>> | ValueDefWithCondition<MarkPropFieldDef<F>>;
+  color?: ChannelDefWithCondition<MarkPropFieldDef<F>>;
```

Comment:
Please do NOT use `ChannelDefWithCondition` in Encoding's declaration because the docs should make it clear whether certain channel support valueDef or not. 

![image](https://user-images.githubusercontent.com/111269/49031365-94c5a200-f15e-11e8-9204-21ec1c4185e1.png)

![image](https://user-images.githubusercontent.com/111269/49031373-998a5600-f15e-11e8-8bf5-9677beb45dd4.png)